app/javascript/packs/timelog.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import Select from 'react-normalized-select'
+
+class Timelog extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { editable: false, card: this.props.timelog.trello_card };
+    this.handleEdit = this.handleEdit.bind(this, this.state);
+  }
+
+  handleEdit() {
+    if(this.state.editable) {
+      var id = this.props.timelog.id;
+      var start_time = this.refs.start_time.value;
+      var duration = this.refs.duration.value;
+      var trello_card = this.state.card;
+      var timelog = {id: id , start_time: start_time , duration: duration, trello_card: trello_card};
+      this.props.handleUpdate(timelog);
+    }
+    this.setState({ editable: !this.state.editable })
+  }
+
+  render() {
+    var start_time = this.state.editable ? <input type='datetime-local' ref='start_time' defaultValue={this.props.timelog.start_time.substring(0, this.props.timelog.start_time.length-5)}/>: <p>Start time: {this.props.timelog.start_time.substring(0, this.props.timelog.start_time.length-5)}</p>;
```

Comment:
It's a mess. Consider moving to separate getter methods. And use formatting for ternary operator